test Trixi with StartUpDG v0.16

jlchan opened the pull request:
Closes #1388

ranocha reviewed (commented):
So there is no breaking change visible to us in Trixi.jl? Do we want to use anything introduced in v0.16 of StartUpDG.jl soon? If not, we could keep compatibility with v0.15 and set the compat section to `StartUpDG = "0.15, 0.16"`?

jlchan commented:
No, apparently there's no breaking change. I added a type parameter to the `Polynomial` approximation type, but that did not seem to affect the existing solvers. 

Eventually, I'd like to replace `GaussSBP` with `Polynomial{Gauss}`, which should make some of the curved mesh extensions cleaner. However, this can be done in another PR. For now, I added `0.15, 0.16` to the compat entry for StartUpDG, and can just remove 0.15 when I make this switch.

ranocha reviewed (approved):
Thanks :+1:
